pkg/resources/kafka/kafka.go

```diff
@@ -211,7 +211,28 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 		}
 	}
 
-	reorderedBrokers := r.reorderBrokers(log, r.KafkaCluster.Spec.Brokers)
+	var brokerPods corev1.PodList
+	matchingLabels := client.MatchingLabels(kafka.LabelsForKafka(r.KafkaCluster.Name))
+	err = r.Client.List(context.TODO(), &brokerPods, client.ListOption(client.InNamespace(r.KafkaCluster.Namespace)), client.ListOption(matchingLabels))
+	if err != nil {
+		return errors.WrapIf(err, "failed to list broker pods that belong to Kafka cluster")
+	}
+
+	var controllerID int32
+	kClient, close, err := r.kafkaClientProvider.NewFromCluster(r.Client, r.KafkaCluster)
+	if err != nil {
+		log.Error(err, "could not create Kafka client, thus could not determine controller")
+	} else {
+		_, controllerID, err = kClient.DescribeCluster()
+		if err != nil {
+			controllerID = -1
+			log.Error(err, "could not find controller broker")
+		}
+
+		defer close()
+	}
```

Comment:
Nit: Can we maybe factor this code out to a method e.g.: func (r *Reconciler) determineControllerId() int32